modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C

```diff
@@ -0,0 +1,271 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AbaqusUMATStress.h"
+#include "Factory.h"
+#include "MooseMesh.h"
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+
+#ifdef LIBMESH_HAVE_DLOPEN
+#include <dlfcn.h>
+#endif
+
+#define QUOTE(macro) stringifyName(macro)
+
+registerMooseObject("TensorMechanicsApp", AbaqusUMATStress);
+
+InputParameters
+AbaqusUMATStress::validParams()
+{
+  InputParameters params = ComputeStressBase::validParams();
+  params.addClassDescription("Coupling material to use Abaqus UMAT models in MOOSE");
+  params.addRequiredParam<FileName>(
+      "plugin", "The path to the compiled dynamic library for the plugin you want to use");
+  params.addRequiredParam<std::vector<Real>>("mechanical_constants",
+                                             "Mechanical Material Properties");
+  params.addParam<std::vector<Real>>("thermal_constants", "Thermal Material Properties");
```

Comment:
This split between mechanical and thermal constants does not make much sense either. They are just put into one vector and passed to the UMAT anyways.